### ISSUE TITLE
Bug: [no-duplicate-imports] Definition for rule '@typescript-eslint/no-duplicate-imports' was not found.

### ISSUE THREAD

**Issue by fengxinming** [typescript-eslint/typescript-eslint#7548]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

none

### Repro Code

```typescript
none
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-duplicate-imports": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

support rule `@typescript-eslint/no-duplicate-imports`

### Actual Result

Definition for rule '@typescript-eslint/no-duplicate-imports' was not found.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/blog/announcing-typescript-eslint-v6#rule-breaking-changes

Please read the release notes when you upgrade.
